improver/utilities/spatial.py

```diff
@@ -100,23 +100,15 @@ def calculate_grid_spacing(
     """
     coord = cube.coord(axis=axis).copy()
     coord.convert_units(units)
-    diffs = np.unique(np.diff(coord.points))
-
-    if len(diffs) > 1:
-        diffs_diff = np.diff(diffs)
-        diffs_mean = np.mean(diffs)
-
-        if np.any(abs(diffs_diff) > rtol * abs(diffs_mean)):
-            raise ValueError(
-                "Coordinate {} points are not equally spaced".format(coord.name())
-            )
-        # use average spacing for considering tiny difference due to rounding error
-        spacing = (coord.points[-1] - coord.points[0]) / (len(coord.points) - 1.0)
+    diffs = np.diff(coord.points)
+    diffs_mean = np.mean(diffs)
 
+    if np.any(abs(diffs - diffs_mean) > rtol * abs(diffs_mean)):
```

Comment:
Use the Numpy function, rather than reinventing it:
```suggestion
    if not np.allclose(diffs, diffs_mean), rtol=rtol):
```

Reply:
Hi Stephen, I looked into https://numpy.org/doc/stable/reference/generated/numpy.allclose.html.  numpy.allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False)'s result is based on "absolute(a - b) <= (atol + rtol * absolute(b)) ". 

With np.any(), I am certain that it is doing what I want ( my personal preference as well).

Reply:
From that description, you would get exactly the same result by setting `atol=0.`

Reply:
Hi Stephen,  I saw that you added atol in the code.  Thanks.